tests/core/test_ls_files.rs

```diff
@@ -0,0 +1,519 @@
+
+    writer.flush()?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::make_namespaces_from;
+
+    use mini_git::core::commands::ls_files::*;
+    use mini_git::core::GitRepository;
+    use mini_git::utils::test::TempDir;
+
+    use std::fs;
+    use std::path::PathBuf;
+    use std::sync::Mutex;
+
+    static FS_MUTEX: Mutex<Option<TempDir<()>>> = Mutex::new(None);
+
+    const GIT_INDEX_PATH: &'static str = const {
+        if cfg!(windows) {
+            ".git\\index"
+        } else {
+            ".git/index"
+        }
+    };
+
+    make_namespaces_from!(make_parser);
+
+    macro_rules! switch_dir {
+        ($body:block) => {
+            match FS_MUTEX.lock() {
+                Ok(inner) if inner.is_some() => {
+                    (inner.as_ref().unwrap()).run(|| $body)
+                }
+                Ok(_) => unreachable!(),
+                Err(..) => panic!("FS Mutex failed!"),
+            }
+        };
+    }
+
+    fn create_temp_repo<'a>() -> TempDir<'a, ()> {
+        let tmp =
+            TempDir::create("cmd_ls_files").with_mutex(&crate::TEST_MUTEX);
+        GitRepository::create(tmp.tmp_dir()).expect("Create repo");
+
+        let test_dir_indicator_file = tmp.tmp_dir().join(".minigit.testdir");
+        fs::File::create(test_dir_indicator_file)
+            .expect("Create test dir indicator file");
+
+        tmp
+    }
+
+    fn setup() {
+        let guard = FS_MUTEX.lock();
+        match guard {
+            Ok(mut inner) if inner.is_none() => {
+                let tmp = create_temp_repo();
+                *inner = Some(tmp);
+            }
+            Ok(..) => {}
+            Err(..) => panic!("Mutex failed!"),
+        };
+    }
+
+    #[test]
+    fn test_create_dummy_git_index() {
+        let temp_file = "test_index";
+        let file_paths =
+            ["file1.txt", "path/to/file2.txt", "another/path/file3.rs"];
+
+        // Create the index file
+        create_dummy_git_index(temp_file, &file_paths).unwrap();
+
+        // Verify file exists with non-zero size
+        let metadata = fs::metadata(temp_file).unwrap();
+        assert!(metadata.len() > 0);
+
+        // Clean up
+        fs::remove_file(temp_file).unwrap();
+    }
+
+    #[test]
+    fn test_ls_files_basic_functionality() {
+        setup();
+        let test_files = ["file1.txt", "src/main.rs", "README.md"];
+        let result = switch_dir!({
+            create_dummy_git_index(GIT_INDEX_PATH, &test_files)
+                .expect("Failed to create dummy index");
+
+            // Test basic ls-files with no arguments
+            let args: [&[&str]; 1] = [&[]];
+            let namespace = make_namespaces(&args).next().unwrap();
+
+            ls_files(&namespace)
+        });
+
+        assert!(result.is_ok(), "ls-files should succeed with basic index");
+
+        let output = result.unwrap();
+        println!("test_ls_files_basic_functionality\n\n--------\n{output:?}--------\n");
```

Comment:
@coderabbitai This is returning
```
test_ls_files_basic_functionality
--------
""--------
```

and similar output for every other test. like the ls_files command just returns no output.
please read src/core/commands/ls_files.rs and see what the issue might be.

Reply:
@coderabbitai these tests literally works perfectly on my local windows machine. so I do not know why exactly it fails in github actions